src/functionaltests/java/com/ericsson/ei/query/QueryAggregatedObjectsTestSteps.java

```diff
@@ -154,49 +153,58 @@ public void perform_invalid_query_on_created_aggregated_object() throws Throwabl
+        queries.add(query2);
 
-        LOGGER.debug("Response of /query RestApi, Status Code: " + response.getStatusCodeValue() +
-                           "\nResponse: " + response.getBody().toString());
+        for(String query: queries) {
+            LOGGER.debug("Freestyle querying for the AggregatedObject with criteria: " + query);
 
-        JsonNode jsonNodeResult = objMapper.readValue(response.getBody().toString(), JsonNode.class);
-        JsonNode aggrObjResponse = objMapper.readValue(jsonNodeResult.get(0).get("aggregatedObject").toString(), JsonNode.class);
+            HttpRequest postRequest = new HttpRequest(HttpMethod.POST);
+            response = postRequest.setPort(applicationPort)
+                    .setHost(hostName)
+                    .addHeader("content-type", "application/json")
+                    .addHeader("Accept", "application/json")
+                    .setEndpoint(entryPoint)
+                    .addParam("request", query)
+                    .performRequest();
 
-        String actualAggrObjId = aggrObjResponse.get("id").asText();
-        LOGGER.debug("AggregatedObject id from Response: " + actualAggrObjId);
+            LOGGER.debug("Response of /query RestApi, Status Code: " + response.getStatusCodeValue() +
+                               "\nResponse: " + response.getBody().toString());
 
+            JsonNode jsonNodeResult = objMapper.readValue(response.getBody().toString(), JsonNode.class);
+            JsonNode aggrObjResponse = objMapper.readValue(jsonNodeResult.get(0).get("aggregatedObject").toString(), JsonNode.class);
 
-        assertEquals(HttpStatus.OK.toString(), Integer.toString(response.getStatusCodeValue()));
-        assertEquals("Failed to compare actual Aggregated Object Id:\n" + actualAggrObjId
-                + "\nwith expected Aggregated Object Id:\n" + expectedAggrId,
-                expectedAggrId, actualAggrObjId);
+            String actualAggrObjId = aggrObjResponse.get("id").asText();
+            LOGGER.debug("AggregatedObject id from Response: " + actualAggrObjId);
+
+
+            assertEquals(HttpStatus.OK.toString(), Integer.toString(response.getStatusCodeValue()));
+            assertEquals("Failed to compare actual Aggregated Object Id:\n" + actualAggrObjId
+                    + "\nwith expected Aggregated Object Id:\n" + expectedAggrId,
+                    expectedAggrId, actualAggrObjId);
+        }
     }
 
     @And("^Perform an invalid freesyle query on Aggregated object$")
     public void perform_invalid_freestyle_query_on_created_aggregated_object() throws Throwable {
         final String invalidAggrId = "6acc3c87-75e0-4b6d-88f5-b1aee4e62b43";
         final String entryPoint = "/query";
-        final String queryAggrObj = "{\"criteria\" :{\"aggregatedObject.id\" : \"" + invalidAggrId + "\" }}";
+        final String queryAggrObj = "{\"criteria\" :{\"aggregatedObject.id\" : \"" + invalidAggrId + "\" }, \"options\": \"\"}";
```

Comment:
remove empty options. I think Andry has fixed to work with missing options entry

Reply:
Thats true, i'll change that!